plugins/snowflake/dbt/include/snowflake/macros/adapters.sql

```diff
@@ -1,14 +1,33 @@
 {% macro snowflake__create_table_as(temporary, relation, sql) -%}
   {%- set transient = config.get('transient', default=true) -%}
+  {%- set cluster_by_keys = config.get('cluster_by', default=none) -%}
+  {%- set enable_automatic_clustering = config.get('automatic_clustering', default=false) -%}
+  {%- if cluster_by_keys is not none and cluster_by_keys is string -%}
+    {%-  set cluster_by_keys = [cluster_by_keys] -%}
+  {%- endif -%}
+  {%- set cluster_by_string = cluster_by_keys|join(", ")-%}
```

Comment:
I think this tries to call `join` on `none` when the `cluster_by` config is unset. Can we try something like:
```
{%- if cluster_by_keys is not none and cluster_by_keys is string -%}
    {%-  set cluster_by_keys = [cluster_by_keys] -%}
    {%- set cluster_by_string = cluster_by_keys|join(", ")-%}
{% else %}
    {%-  set cluster_by_keys = [] -%}
    {%- set cluster_by_string = ""-%}
{%- endif -%}

Reply:
and then below, we'll want to do:
```
{% if cluster_by_string is not none -%}
    alter table {{relation}} cluster by ({{cluster_by_string}});
{%- endif -%}
```

Reply:
Oh damn, that was really stupid of me... 

But actually the whole logic is pretty wild here... I was doing really stupid stuff but it should work after my last commit. Let me know if you don't agree with that flow. I tested it and it works both for list of strings or just one string.

I also noticed a corner case. We're allowing `alter table {{relation}} resume recluster;` to run on the condition that this config argument be set to true. The issue with that is that if the table was not created as a clustered table (i.e. that no `cluster_by_keys` were provided) then Snowflake will throw an error that the table is not clustered. 

I changed this piece of logic as well. Let me know what you think.